interfaces/repo.go

```diff
@@ -970,7 +1003,7 @@ func (r *Repository) AutoConnectCandidatePlugs(slotSnapName, slotName string, po
 			plug := &Plug{PlugInfo: plugInfo}
 			slot := &Slot{SlotInfo: slotInfo}
 			// declaration based checks disallow
-			if !policyCheck(plug, slot) {
+			if !policyCheck(NewConnectedPlug(plugInfo, nil), NewConnectedSlot(slotInfo, nil)) {
```

Comment:
Should we pay attention to making sure that connected plug / slot objects are only created in a controlled way and kept somewhere in the repository/daemon/wherever (so that there's one such object per connection).